### PR TITLE
[Merged by Bors] - fix bevy_ecs macro path handling

### PR DESCRIPTION
**Pull request by jakobhellermann**
- It now doesn't search in the dev-dependencies anymore
- and the behaviour is consistent for derive_bundle and derive_system_param

### CODE REVIEW

**Comment by DJMcNab** [comment]
How does this work if a user only uses the procedural macro in tests or examples?

**Comment by jakobhellermann** [comment]
The use case where I ran into this was adding `bevy` to `bevy_app`'s dev-dependencies, so that we can `use bevy::prelude::*;` in doc tests. The macro saw `bevy` in the dev-dependencies so tried to use that, but doctests don't have access to dev-dependencies. 

The only way it would break now is if someone has `bevy` (or `bevy_ecs`) in their dev-dependencies but not in the main dependencies, which I think is pretty unlikely.

**Comment by jakobhellermann** [comment]
But to support that we could first try in regular dependencies and afterwards the dev-dependencies. Should I do that?

**Comment by DJMcNab** [comment]
I think so? I'm getting a massive sense of deja-vu - that is, I feel like we've already made this change somewhere else already, although I can't find it now.

But yes, trying in regular dependencies first and then dev dependencies does make sense.

**Comment by MinerSebas** [comment]
In #1182 there was a similar change made to the reflect crate.

**Comment by jakobhellermann** [comment]
I changed it. I think `build-dependencies` can be ignored :smile:

File: crates/bevy_ecs/macros/src/lib.rs [hunk 1]
```diff
@@ -54,17 +54,8 @@ fn derive_bundle_(input: DeriveInput) -> Result<TokenStream2> {
...
-    } else if let Some(package) = manifest.find(|name| name == "bevy_internal") {
-        format!("{}::ecs", package.name)
-    } else if let Some(package) = manifest.find(|name| name == "bevy_ecs") {
-        package.name
-    } else {
-        "bevy_ecs".to_string()
-    };
-    let crate_path: Path = syn::parse(path_str.parse::<TokenStream>().unwrap()).unwrap();
+
+    let crate_path = bevy_ecs_path();
```
Comment (DJMcNab) [inline crates/bevy_ecs/macros/src/lib.rs:58]: Maybe `ecs_crate` or even just `ecs` would be better?

File: crates/bevy_ecs/macros/src/lib.rs [hunk 2]
```diff
@@ -443,26 +428,49 @@ pub fn derive_system_param(input: TokenStream) -> TokenStream {
...
-                    #(#fields: <<#field_types as SystemParam>::Fetch as #path::FetchSystemParam>::get_param(system_state, world, resources)?,)*
+                    #(#fields: <<#field_types as SystemParam>::Fetch as #crate_path::FetchSystemParam>::get_param(system_state, world, resources)?,)*
                     #(#ignored_fields: <#ignored_field_types>::default(),)*
                 })
             }
         }
     })
 }
+
+fn bevy_ecs_path() -> Path {
```
Comment (DJMcNab) [inline crates/bevy_ecs/macros/src/lib.rs:455]: I don't like that `Path` is overloaded here, but that's `syn`'s fault not yours

**Review by DJMcNab — APPROVED** [review]
Looks good. One small suggestion cart might want to merge, but equally this can also be merged as is.

File: crates/bevy_ecs/macros/src/lib.rs (hunk 1 above)
Comment (jakobhellermann) [reply crates/bevy_ecs/macros/src/lib.rs]: Changed it to `ecs`.

File: crates/bevy_ecs/macros/src/lib.rs (hunk 2 above)
Comment (jakobhellermann) [reply crates/bevy_ecs/macros/src/lib.rs]: I qualified the Path as `syn::Path`, I agree that reusing `Path` is confusing here.

**Comment by DJMcNab** [comment]
Don't fix the clippy failures in this PR please - someone else will do that soon

File: crates/bevy_ecs/macros/src/lib.rs (hunk 1 above)
Comment (cart) [reply crates/bevy_ecs/macros/src/lib.rs]: imo removing "path" from the name makes it harder to tell what the variable is. can we do `ecs_path`?

Comment (cart) [reply crates/bevy_ecs/macros/src/lib.rs]: or alternatively ecs_crate?

Comment (jakobhellermann) [reply crates/bevy_ecs/macros/src/lib.rs]: I changed it to `ecs_path`.

**Comment by woubuc** [comment]
I don't know if you caught this but I added a couple of missing paths to the `SystemParam` macro in #1434 the other day. It looks like those are still missing in this PR.

**Comment by jakobhellermann** [comment]
Thanks, I added them now.

**Review by DJMcNab — APPROVED** [review]
Everything looks good

**Comment by cart** [comment]
resolve merge conflicts and I think this is good to go!

**Review by cart — APPROVED** [review]

**Comment by jakobhellermann** [comment]
done

**Pull request closed without merging**